tests/ChatTests/Local.hs

```diff
@@ -0,0 +1,128 @@
+  alice <# "- Greetings."
+
+testUserNotes :: FilePath -> IO ()
+testUserNotes tmp = withNewTestChat tmp "alice" aliceProfile $ \alice -> do
+  alice /- "keep in mind"
+  alice ##> "/tail"
+  alice <# "- keep in mind"
+
+  alice ##> "/create user secret"
+  alice <## "user profile: secret"
+  alice <## "use /p <display name> to change it"
+  alice <## "(the updated profile will be sent to all your contacts)"
+
+  alice ##> "/tail"
+
+  alice ##> "/_delete item $1 1 internal"
+  alice <## "chat db error: SENoteFolderNotFound {noteFolderId = 1}"
+
+testPreviewsPagination :: FilePath -> IO ()
+testPreviewsPagination tmp = withNewTestChat tmp "alice" aliceProfile $ \alice -> do
+  tsS <- iso8601Show <$> getCurrentTime
+  alice /- "first"
+  tsM <- iso8601Show <$> getCurrentTime
+  alice /- "last"
+  tsE <- iso8601Show <$> getCurrentTime
+
+  -- there's only one folder that got updated after tsM and before tsE
+  getChats_ alice "count=3" [("-", "last")]
+  getChats_ alice ("after=" <> tsE <> " count=10") []
+  getChats_ alice ("after=" <> tsS <> " count=10") [("-", "last")]
+  getChats_ alice ("before=" <> tsM <> " count=10") []
+  getChats_ alice ("before=" <> tsE <> " count=10") [("-", "last")]
+  getChats_ alice ("before=" <> tsS <> " count=10") []
+
+testFiles :: FilePath -> IO ()
+testFiles tmp = withNewTestChat tmp "alice" aliceProfile $ \alice -> do
+  -- setup
+  let files = "./tests/tmp/app_files"
+  alice ##> ("/_files_folder " <> files)
+  alice <## "ok"
+
+  -- ui-like upload
+  let source = "./tests/fixtures/test.jpg"
+  let stored = files </> "test.jpg"
+  copyFile source stored
+  alice ##> "/_create $1 json {\"filePath\": \"test.jpg\", \"msgContent\": {\"text\":\"hi myself\",\"type\":\"image\",\"image\":\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAgAAAAIAQMAAAD+wSzIAAAABlBMVEX///+/v7+jQ3Y5AAAADklEQVQI12P4AIX8EAgALgAD/aNpbtEAAAAASUVORK5CYII=\"}}"
+  alice <# "- hi myself"
+  alice <# "- file 1 (test.jpg)"
+
+  alice ##> "/tail"
+  alice <# "- hi myself"
+  alice <# "- file 1 (test.jpg)"
+
+  alice ##> "/_get chat $1 count=100"
+  r <- chatF <$> getTermLine alice
+  r `shouldBe` [((1, "hi myself"), Just "test.jpg")]
+
+  alice ##> "/fs 1"
+  alice <## "local file 1 (test.jpg)"
+
+  -- one more file
+  let stored2 = files </> "another_test.jpg"
+  copyFile source stored2
+  alice ##> "/_create $1 json {\"filePath\": \"another_test.jpg\", \"msgContent\": {\"text\":\"\",\"type\":\"image\",\"image\":\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAgAAAAIAQMAAAD+wSzIAAAABlBMVEX///+/v7+jQ3Y5AAAADklEQVQI12P4AIX8EAgALgAD/aNpbtEAAAAASUVORK5CYII=\"}}"
+  alice <# "- file 2 (another_test.jpg)"
+
+  alice ##> "/_delete item $1 2 internal"
+  alice <## "message deleted"
+  doesFileExist stored2 `shouldReturn` False
+  doesFileExist stored `shouldReturn` True
+
+  alice ##> "/clear -"
+  alice ##> "/fs 1"
+  alice <## "chat db error: SEChatItemNotFoundByFileId {fileId = 1}"
+  alice ##> "/tail"
+  doesFileExist stored `shouldReturn` False
```

Comment:
also test that that clearing notes chat doesn't remove files from other conversations (it does use some new function to retrieve list of files I think). maybe it's better to split this second part into one or two separate tests to easier identify what fails when something breaks tests.